middleware/privilegetoken/main_test.go

```diff
@@ -0,0 +1,231 @@
+package privilegetoken
+
+import (
+	"fmt"
+	"log"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/gofiber/fiber/v2"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/rs/zerolog"
+	"github.com/stretchr/testify/assert"
+)
+
+type httpTestTemplate struct {
+	description  string // description of the test case
+	route        string // route path to test
+	expectedCode int    // expected HTTP status code
+}
+
+type testHelper struct {
+	app                 *fiber.App
+	t                   *testing.T
+	assert              *assert.Assertions
+	logger              zerolog.Logger
+	privilegeMiddleware IVerifyPrivilegeToken
+}
+
+func initTestHelper(t *testing.T) testHelper {
+	assert := assert.New(t)
+	app := fiber.New() // This can be moved into a new var to avoid re-initializing TBD.
+	logger := zerolog.New(os.Stdout).With().Timestamp().Logger()
+
+	return testHelper{
+		assert: assert,
+		t:      t,
+		app:    app,
+		logger: logger,
+		privilegeMiddleware: New(Config{
+			Log: &logger,
+		}),
+	}
+}
+
+func (t testHelper) signToken(p jwt.MapClaims) *jwt.Token {
+	return jwt.NewWithClaims(jwt.SigningMethodHS256, p)
+}
+
+const (
+	Commands        = 1
+	AllTimeLocation = 2
+)
+
+func TestSuccessOnValidSinglePrivilege(t *testing.T) {
+	th := initTestHelper(t)
+
+	test := httpTestTemplate{
+		description:  "Test success response when token contains at only allowed privilege on endpoint",
+		route:        fmt.Sprintf("/v1/test/%d", Commands),
```

Comment:
`Commands` seems a bit misleading here, this is the token id, not a privilege id.